doc/contribute.rst

```diff
@@ -44,6 +45,24 @@ Submit Pull Request
   - Fix the problems reported by automatic checks
   - If you are contributing a new module, consider add a testcase in `tests <https://github.com/dmlc/xgboost/tree/master/tests>`_.
 
+*********************************
+Running Formatting Checks Locally
+*********************************
+
+Once you submit a pull request to `dmlc/xgboost <https://github.com/dmlc/xgboost>`_, we perform
+automatic checks to enforce coding style conventions (we use `pylint <https://github.com/PyCQA/pylint>`_,
+`cpplint <https://github.com/cpplint/cpplint>`_, and `clang-tidy <https://clang.llvm.org/extra/clang-tidy/>`_).
+You can run formatting checks locally using the following commands (Docker required):
+
+.. code-block:: bash
+
+  tests/ci_build/ci_build.sh clang_tidy docker --build-arg CUDA_VERSION=9.2 tests/ci_build/clang_tidy.sh
```

Comment:
We already have documentation for tidy in this file.

I don't think it takes a docker to run the lint for normal contributors.  It's the safest way to go, but maybe add a note for running it without docker and just let the command line error report whatever is missing?

Reply:
The CI server will run clang-tidy with both `--cpp` and `--cuda` enabled, and for `--cuda`, we need the CUDA toolkit. (Don't need a physical GPU however.)

Reply:
@trivialfis Let me think over a bit and add a sentence about running the checks without Docker. My original intention was to exactly reproduce whatever the Jenkins CI server is doing, but I see how Docker may be too high requirement for some contributors.